modules/db/src/main/scala/gem/dao/StepDao.scala

```diff
@@ -141,22 +143,24 @@ object StepDao {
     sql"""
       SELECT s.instrument,
              s.step_type,
-             sg.continuum,
-             sg.ar_arc,
-             sg.cuar_arc,
-             sg.thar_arc,
-             sg.xe_arc,
-             sg.filter,
-             sg.diffuser,
-             sg.shutter,
-             sg.exposure_time,
-             sg.coadds,
+             gc.continuum,
+             gc.ar_arc,
+             gc.cuar_arc,
+             gc.thar_arc,
+             gc.xe_arc,
+             gc.filter,
+             gc.diffuser,
+             gc.shutter,
+             gc.exposure_time,
+             gc.coadds,
              sc.offset_p,
              sc.offset_q,
              ss.type
         FROM step s
              LEFT OUTER JOIN step_gcal sg
```

Comment:
It worries me too.  I don't think this is maintainable going forward.  I think this was just done as an initial experiment.

Going forward I assume we'll need two queries.  The first will determine the step type and then the second will read the information specific to that step type.  Already we aren't really doing a "selectAll" -- note that only the instrument name is included not all the instrument configuration.  It might be feasible with just one instrument, but with just a handful we won't be able to keep this implementation.